sklearn/linear_model/cd_fast.pyx

```diff
@@ -289,8 +289,9 @@ def enet_coordinate_descent(np.ndarray[DOUBLE, ndim=1] w,
 @cython.cdivision(True)
 def sparse_enet_coordinate_descent(double[:] w,
                             double alpha, double beta,
-                            double[:] X_data, int[:] X_indices,
-                            int[:] X_indptr, double[:] y,
+                            np.ndarray[double, ndim=1] X_data,
+                            np.ndarray[int, ndim=1] X_indices,
+                            np.ndarray[int, ndim=1] X_indptr, np.ndarray[double, ndim=1] y,
```

Comment:
> See PR #4775 on which this PR is based : it does not affect performance.

OK, sorry for asking. I am lost in everything that is happening.

> I should have made this PR independant, should I remove the commits
> related to PR #4775 ?

In general it's good to keep PRs separated. But here I find that the
diffs are still very readable, so to be pragmatic, I wouldn't loose any
time on changing things.